### PR TITLE
fix: page not emitted when columns are sorted

### PR DESCRIPTION
**Pull request by chintankavathia**
Allows to keep external pagination components to be in sync with current page value.

**What kind of change does this PR introduce?** (check one with "x")

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here) 
external pagination component will not update the page number when user sort the columns

**What is the new behavior?**
external pagination component's page number stays in sync with the datatable offset value.

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by chintankavathia** [comment]
@timowolf Could you please review this one?

File: projects/ngx-datatable/src/lib/components/datatable.component.ts [hunk 1]
```diff
@@ -1162,6 +1162,13 @@ export class DatatableComponent implements OnInit, DoCheck, AfterViewInit, After
 
     // Always go to first page when sorting to see the newly sorted data
     this.offset = 0;
+    // Also emit the page object
+    this.page.emit({
+      count: this.count,
+      pageSize: this.pageSize,
+      limit: this.limit,
+      offset: this.offset
+    });
```
Comment (timowolf) [inline projects/ngx-datatable/src/lib/components/datatable.component.ts:1171]: I would move it below `this.bodyComponent.updateOffsetY(this.offset);`. So internals are computed first. Also consistent with ` onFooterPage(event: any)`.

**Review by timowolf — CHANGES_REQUESTED** [review]

**Review by timowolf — APPROVED** [review]
Many thanks. Looks good to me.

**Comment by Hrudwolf** [comment]
I've noticed an issue following the integration of PR #45, particularly concerning the interaction between server-side paging and sorting. When sorting a column, the application makes two server calls: one for paging and another for sorting. 

With this change, there is no distinction between retrieving page data and applying sorting.
And because the `sort.emit` is triggered after the `page.emit` we cannot apply the sort to getting the page data.

A suggested fix would be to swap the page and sort emits around or not call `page.emit` if `externalPaging` is `true`.

**Comment by chintankavathia** [comment]
> I've noticed an issue following the integration of PR #45, particularly concerning the interaction between server-side paging and sorting. When sorting a column, the application makes two server calls: one for paging and another for sorting.
> 
> With this change, there is no distinction between retrieving page data and applying sorting. And because the `sort.emit` is triggered after the `page.emit` we cannot apply the sort to getting the page data.
> 
> A suggested fix would be to swap the page and sort emits around or not call `page.emit` if `externalPaging` is `true`.

The order here is correct. sort should trigger only after the page offset is reset to 0. Yes in case of external paging it can lead to multiple server calls but that can easily be avoided using `switchMap`.

**Comment by amelnikoff** [comment]
So for anyone else finding this PR, safe version is [22.1.0]

**Comment by amelnikoff** [comment]
Nah, this is suffering from using old open source project. Siemens team has same problem ;) I don't think that their managers supporting development of OSS.

**Comment by timowolf** [comment]
@talamaska @amelnikoff we hear you, but priorized a general rework of the project structure first.

**Comment by talamaska** [comment]
We are listening to both events. In what world sorting changes the page? No matter if it is backend or just front end, assuming the user wanted to change the page when sorting is not supported by any real UAT. Again the change was introduced as a minor release, which is unacceptable. There is no way to opt-out this new behaviour. The suggested approach with event source is ok-ish if it will help us prevent complicated rxjs logic to avoid duplicated requests. We are performing backend sorting and pagination, I did't quite understand what are @timowolf saying about this change being made for exactly this. Hopefully this will be resolved soon.

**Comment by spike-rabbit** [comment]
FYI, in v23 the page event now also contains `sorts` which turns this effectively into the external / server state change event we were planning. 

Since sorting will always reset the page and thus emit a page event, this should allow lazy data loading by only listening to one event. Note that the page event is still emitted on sort, even if the page itself was not actually changed.

**Pull request merged**